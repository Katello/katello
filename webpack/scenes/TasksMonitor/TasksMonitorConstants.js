export const TASKS_MONITOR_STARTED = 'TASKS_MONITOR_STARTED';
export const TASKS_MONITOR_STOPPED = 'TASKS_MONITOR_STOPPED';
export const TASKS_MONITOR_SUCCESS = 'TASKS_MONITOR_SUCCESS';
export const TASKS_MONITOR_FAILED = 'TASKS_MONITOR_FAILED';
export const TASKS_MONITOR_ALREADY_MONITORED = 'TASKS_MONITOR_ALREADY_MONITORED';
