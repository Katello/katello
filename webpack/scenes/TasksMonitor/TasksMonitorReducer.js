import Immutable from 'seamless-immutable';
import {
  TASKS_MONITOR_STARTED,
  TASKS_MONITOR_STOPPED,
  TASKS_MONITOR_SUCCESS,
  TASKS_MONITOR_FAILED,
} from './TasksMonitorConstants';

const initialState = Immutable({ });

export default (state = initialState, action) => {
  switch (action.type) {
    case TASKS_MONITOR_STARTED:
      return state.setIn([action.payload.id], {
        interval: action.payload.interval,
        intervalId: action.payload.intervalId,
        params: action.payload.params,
        tasks: [],
      });
    case TASKS_MONITOR_STOPPED:
      return state.setIn([action.payload.id, 'intervalId'], null);

    case TASKS_MONITOR_SUCCESS:
      return state
        .setIn([action.payload.id, 'tasks'], action.payload.tasks)
        .setIn([action.payload.id, 'error'], null);

    case TASKS_MONITOR_FAILED:
      return state.setIn([action.payload.id, 'error'], action.payload.error);

    default:
      return state;
  }
};
