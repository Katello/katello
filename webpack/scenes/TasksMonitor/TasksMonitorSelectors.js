export const selectTasksMonitorState = state => state.katello.tasksMonitor;

export const selectMonitor = (state, monitorId) =>
  selectTasksMonitorState(state)[monitorId];

export const selectMonitorTasks = (state, monitorId) => {
  const monitor = selectMonitor(state, monitorId);

  return monitor && monitor.tasks;
};

export const selectIsMonitorActive = (state, monitorId) => {
  const monitor = selectMonitor(state, monitorId);

  return monitor && !!monitor.intervalId;
};
