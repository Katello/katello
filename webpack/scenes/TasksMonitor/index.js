import reducer from './TasksMonitorReducer';

// eslint-disable-next-line import/prefer-default-export
export const reducers = { tasksMonitor: reducer };
