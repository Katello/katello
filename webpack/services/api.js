import { getApiRequest } from './mock';

// eslint-disable-next-line import/prefer-default-export
export const get = ({ name }) =>
  new Promise((resolve, reject) => {
    const request = getApiRequest(name);

    if (request) {
      return resolve(request.response);
    }

    return reject(new Error('No api request found with that name.'));
  });
