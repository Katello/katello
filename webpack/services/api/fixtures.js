export default [
  {
    path: '/katello/api/v2/products/:id/repository_sets/:repo_set_id/enable',
    searchRegex: /\/katello\/api\/v2\/products\/\d+\/repository_sets\/\d+\/enable/,
    type: 'PUT',
    response: () => {
      const types = ['yum', 'source_rpm', 'debug', 'iso', 'beta', 'kickstart'];

      const output = {
        id: Math.round(Math.random() * 500),
        name: 'Red Hat Enterprise Linux 6.2 Server for RHS 2 VSA Beta RPMs from RHUI x86_64',
        content_type: types[Math.floor(Math.random() * types.length)],
      };

      return [
        200,
        {
          id: 'd4a4432e-4987-4206-aa99-615b5b15fbf8',
          label: 'Actions::Katello::RepositorySet::EnableRepository',
          pending: false,
          username: 'admin',
          started_at: '2017-12-19 14:33:41 UTC',
          ended_at: '2017-12-19 14:33:44 UTC',
          state: 'stopped',
          result: 'success',
          progress: 1.0,
          input: {
            services_checked: ['pulp', 'pulp_auth', 'candlepin', 'candlepin_auth'],
            repository: {
              id: 7,
              name: 'Red Hat Enterprise Linux 6.2 Server for RHS 2 VSA Beta RPMs from RHUI x86_64',
              label: 'Red_Hat_Enterprise_Linux_6_2_Server_for_RHS_2_VSA_Beta_RPMs_from_RHUI_x86_64',
            },
            product: {
              id: 155,
              name: 'Red Hat Storage',
              label: 'Red_Hat_Storage',
              cp_id: '171',
            },
            provider: {
              id: 2,
              name: 'Red Hat',
            },
            organization: {
              id: 1,
              name: 'Default Organization',
              label: 'Default_Organization',
            },
          },
          output,
          humanized: {
            action: 'Enable',
            input: [],
            output: '',
            errors: [],
          },
          cli_example: null,
        },
      ];
    },
  },
  {
    path: '/katello/api/v2/products/:id/repository_sets/:repo_set_id/disable',
    searchRegex: /\/katello\/api\/v2\/products\/\d+\/repository_sets\/\d+\/disable/,
    type: 'PUT',
    response: () => [
      200,
      {
        id: 'd4a4432e-4987-4206-aa99-615b5b15fbf8',
        label: 'Actions::Katello::RepositorySet::DisableRepository',
        pending: false,
        username: 'admin',
        started_at: '2017-12-19 14:33:41 UTC',
        ended_at: '2017-12-19 14:33:44 UTC',
        state: 'stopped',
        result: 'success',
        progress: 1.0,
        input: {
          services_checked: ['pulp', 'pulp_auth', 'candlepin', 'candlepin_auth'],
          repository: {
            id: 7,
            name: 'Red Hat Enterprise Linux 6.2 Server for RHS 2 VSA Beta RPMs from RHUI x86_64',
            label: 'Red_Hat_Enterprise_Linux_6_2_Server_for_RHS_2_VSA_Beta_RPMs_from_RHUI_x86_64',
          },
          product: {
            id: 155,
            name: 'Red Hat Storage',
            label: 'Red_Hat_Storage',
            cp_id: '171',
          },
          provider: {
            id: 2,
            name: 'Red Hat',
          },
          organization: {
            id: 1,
            name: 'Default Organization',
            label: 'Default_Organization',
          },
        },
        output: {},
        humanized: {
          action: 'Disable',
          input: [],
          output: '',
          errors: [],
        },
        cli_example: null,
      },
    ],
  },
  {
    path: '/organizations/:id/repository_sets',
    searchRegex: /\/organizations\/\d+\/repository_sets$/,
    response: () => [
      200,
      {
        total: 5,
        subtotal: 5,
        page: null,
        per_page: null,
        error: null,
        search: null,
        sort: {
          by: null,
          order: null,
        },
        results: [
          {
            repositories: [],
            product: {
              name: 'Red hat Enterprise Linux Server 7',
              id: 5,
            },
            type: 'file',
            vendor: 'Red Hat',
            gpgUrl: 'http://',
            contentUrl: '/content/dist/rhel/server/7///source/iso',
            id: '2457',
            name: 'Red Hat Enterprise Linux 7 Server (Source ISOs)',
            label: 'rhel-7-server-source-isos',
          },
          {
            repositories: [],
            type: 'yum',
            product: {
              name: 'Red hat Enterprise Linux Server 7',
              id: 5,
            },
            vendor: 'Red Hat',
            gpgUrl: 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release',
            contentUrl: '/content/dist/rhel/server/6/6Server//sat-tools/6.1/source/SRPMS',
            id: '4187',
            name: 'Red Hat Satellite Tools 6.1 (for RHEL 6 Server) (Source RPMs)',
            label: 'rhel-6-server-satellite-tools-6.1-source-rpms',
          },
          {
            repositories: [],
            type: 'yum',
            product: {
              name: 'Red hat Enterprise Linux Server 7',
              id: 5,
            },
            vendor: 'Red Hat',
            gpgUrl:
              'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-beta,file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release',
            contentUrl: '/content/beta/rhel/server/6/6Server//rh-common/os',
            id: '3076',
            name: 'Red Hat Enterprise Linux 6 Server - RH Common Beta (RPMs)',
            label: 'rhel-6-server-rh-common-beta-rpms',
          },
          {
            repositories: [],
            type: 'yum',
            product: {
              name: 'Red hat Enterprise Linux Server 7',
              id: 5,
            },
            vendor: 'Red Hat',
            gpgUrl: 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release',
            contentUrl: '/content/dist/rhel/server/6///rhev-agent/3/os',
            id: '1699',
            name: 'Red Hat Enterprise Virtualization Agents for RHEL 6 Server (RPMs)',
            label: 'rhel-6-server-rhev-agent-rpms',
          },
          {
            repositories: [
              {
                id: 631,
                name: 'Red Hat Enterprise Linux 6 Server Kickstart x86_64 6.8',
                releasever: '6.8',
                arch: 'x86_64',
              },
            ],
            product: {
              name: 'Red hat Enterprise Linux Server 7',
              id: 5,
            },
            type: 'kickstart',
            vendor: 'Red Hat',
            gpgUrl: 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release',
            contentUrl: '/content/dist/rhel/server/6///kickstart',
            id: '1952',
            name: 'Red Hat Enterprise Linux 6 Server (Kickstart)',
            label: 'rhel-6-server-kickstart',
          },
        ],
      },
    ],
  },
  {
    // Returns the currently enabled repositories
    path: '/repository_sets?organization_id=:id',
    searchRegex: /\/organizations\/\d+\/repository_sets/,
    response: () => [
      200,
      {
        total: 5,
        subtotal: 5,
        page: null,
        per_page: null,
        error: null,
        search: null,
        sort: {
          by: null,
          order: null,
        },
        results: [
          {
            repositories: [
              {
                id: 631,
                name: 'Red Hat Enterprise Linux 6 Server Kickstart x86_64 6.8',
                releasever: '6.8',
                arch: 'x86_64',
              },
            ],
            product: {
              name: 'Red hat Enterprise Linux Server 7',
              id: 5,
            },
            type: 'kickstart',
            vendor: 'Red Hat',
            gpgUrl: 'file:///etc/pki/rpm-gpg/RPM-GPG-KEY-redhat-release',
            contentUrl: '/content/dist/rhel/server/6///kickstart',
            id: '1952',
            name: 'Red Hat Enterprise Linux 6 Server (Kickstart)',
            label: 'rhel-6-server-kickstart',
          },
        ],
      },
    ],
  },
  {
    path: '/products/:product_id/repository_sets/:content_id/available_repositories',
    searchRegex: /\/products\/\d+\/repository_sets\/\d+\/available_repositories/,
    response: () => [
      200,
      {
        total: 6,
        subtotal: 6,
        page: null,
        per_page: null,
        error: null,
        search: null,
        sort: {
          by: null,
          order: null,
        },
        results: [
          {
            substitutions: {
              releasever: '7.0',
              basearch: 'x86_64',
            },
            path: '/content/dist/rhel/server/7/7.0/x86_64/supplementary/os',
            repo_name: 'Red Hat Enterprise Linux 7 Server - Supplementary RPMs x86_64 7.0',
            name: 'Red Hat Enterprise Linux 7 Server - Supplementary (RPMs)',
            pulp_id:
              'Default_Organization-Red_Hat_Enterprise_Linux_Server-Red_Hat_Enterprise_Linux_7_Server_-_Supplementary_RPMs_x86_64_7_0',
            enabled: false,
            promoted: false,
          },
          {
            substitutions: {
              releasever: '7.1',
              basearch: 'x86_64',
            },
            path: '/content/dist/rhel/server/7/7.1/x86_64/supplementary/os',
            repo_name: 'Red Hat Enterprise Linux 7 Server - Supplementary RPMs x86_64 7.1',
            name: 'Red Hat Enterprise Linux 7 Server - Supplementary (RPMs)',
            pulp_id:
              'Default_Organization-Red_Hat_Enterprise_Linux_Server-Red_Hat_Enterprise_Linux_7_Server_-_Supplementary_RPMs_x86_64_7_1',
            enabled: false,
            promoted: false,
          },
          {
            substitutions: {
              releasever: '7.2',
              basearch: 'x86_64',
            },
            path: '/content/dist/rhel/server/7/7.2/x86_64/supplementary/os',
            repo_name: 'Red Hat Enterprise Linux 7 Server - Supplementary RPMs x86_64 7.2',
            name: 'Red Hat Enterprise Linux 7 Server - Supplementary (RPMs)',
            pulp_id:
              'Default_Organization-Red_Hat_Enterprise_Linux_Server-Red_Hat_Enterprise_Linux_7_Server_-_Supplementary_RPMs_x86_64_7_2',
            enabled: false,
            promoted: false,
          },
          {
            substitutions: {
              releasever: '7.3',
              basearch: 'x86_64',
            },
            path: '/content/dist/rhel/server/7/7.3/x86_64/supplementary/os',
            repo_name: 'Red Hat Enterprise Linux 7 Server - Supplementary RPMs x86_64 7.3',
            name: 'Red Hat Enterprise Linux 7 Server - Supplementary (RPMs)',
            pulp_id:
              'Default_Organization-Red_Hat_Enterprise_Linux_Server-Red_Hat_Enterprise_Linux_7_Server_-_Supplementary_RPMs_x86_64_7_3',
            enabled: false,
            promoted: false,
          },
          {
            substitutions: {
              releasever: '7.4',
              basearch: 'x86_64',
            },
            path: '/content/dist/rhel/server/7/7.4/x86_64/supplementary/os',
            repo_name: 'Red Hat Enterprise Linux 7 Server - Supplementary RPMs x86_64 7.4',
            name: 'Red Hat Enterprise Linux 7 Server - Supplementary (RPMs)',
            pulp_id:
              'Default_Organization-Red_Hat_Enterprise_Linux_Server-Red_Hat_Enterprise_Linux_7_Server_-_Supplementary_RPMs_x86_64_7_4',
            enabled: false,
            promoted: false,
          },
          {
            substitutions: {
              releasever: '7Server',
              basearch: 'x86_64',
            },
            path: '/content/dist/rhel/server/7/7Server/x86_64/supplementary/os',
            repo_name: 'Red Hat Enterprise Linux 7 Server - Supplementary RPMs x86_64 7Server',
            name: 'Red Hat Enterprise Linux 7 Server - Supplementary (RPMs)',
            pulp_id:
              'Default_Organization-Red_Hat_Enterprise_Linux_Server-Red_Hat_Enterprise_Linux_7_Server_-_Supplementary_RPMs_x86_64_7Server',
            enabled: false,
            promoted: false,
          },
        ],
      },
    ],
  },
];
