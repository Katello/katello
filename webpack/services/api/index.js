import axios from 'axios';
import Immutable from 'seamless-immutable';

const getcsrfToken = () => {
  const token = document.querySelector('meta[name="csrf-token"]');
  return token ? token.content : '';
};

axios.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest';
axios.defaults.headers.common['X-CSRF-Token'] = getcsrfToken();

export const initialApiState = Immutable({
  loading: true,
  pagination: {
    page: 0,
    perPage: 20,
  },
  itemCount: 0,
  results: [],
});

class Api {
  constructor() {
    this.baseApiPath = '/katello/api/v2';
  }

  getApiUrl(url) {
    return this.baseApiPath + url;
  }

  get(url, headers = {}, params = {}) {
    return axios.get(this.getApiUrl(url), {
      headers,
      params,
    });
  }

  put(url, data = {}, headers = {}) {
    return axios.put(this.getApiUrl(url), data, {
      headers,
    });
  }

  post(url, data = {}, headers = {}) {
    return axios.post(this.getApiUrl(url), data, {
      headers,
    });
  }

  delete(url, headers = {}, data = {}) {
    return axios.delete(this.getApiUrl(url), {
      headers,
      data,
    });
  }

  patch(url, data = {}, headers = {}) {
    return axios.patch(this.getApiUrl(url), data, {
      headers,
    });
  }
}

export default new Api();

class ForemanTasksApi extends Api {
  constructor() {
    super();
    this.baseApiPath = '/foreman_tasks/api';
  }
}

export const foremanTasksApi = new ForemanTasksApi();

// eslint-disable-next-line import/prefer-default-export
const orgNode = document.getElementById('organization-id');
// This node does not exist while testing
export const orgId = orgNode ? orgNode.dataset.id : '1';
