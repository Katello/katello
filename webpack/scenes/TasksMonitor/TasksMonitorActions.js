import { foremanTasksApi as api } from '../../services/api';
import { propsToSnakeCase } from '../../services/index';

import {
  TASKS_MONITOR_STARTED,
  TASKS_MONITOR_STOPPED,
  TASKS_MONITOR_SUCCESS,
  TASKS_MONITOR_FAILED,
  TASKS_MONITOR_ALREADY_MONITORED,
} from './TasksMonitorConstants';
import { selectMonitor, selectIsMonitorActive } from './TasksMonitorSelectors';

const isUnauthorizedResponse = apiResponse => apiResponse.status === 401;

const performBulkSearch = (params = {}) => {
  const apiParams = {
    searches: [{ ...propsToSnakeCase(params) }],
  };

  return api.post('/tasks/bulk_search', apiParams);
};

export const stopMonitoringTasks = id => (dispatch, getState) => {
  const { intervalId } = selectMonitor(getState(), id);

  clearInterval(intervalId);

  dispatch({
    type: TASKS_MONITOR_STOPPED,
    payload: { id, intervalId },
  });
};

export const runMonitorLifecycle = id => async (dispatch, getState) => {
  try {
    const { params } = selectMonitor(getState(), id);
    const response = await performBulkSearch(params);

    if (isUnauthorizedResponse(response)) {
      dispatch(stopMonitoringTasks(id));
      throw new Error('Unauthorized');
    }

    const tasks = response.data[0].results;

    dispatch({
      type: TASKS_MONITOR_SUCCESS,
      payload: { id, tasks },
    });
  } catch (error) {
    dispatch({
      type: TASKS_MONITOR_FAILED,
      payload: { id, error },
    });
  }
};

export const startMonitoringTasks = ({ id = 'some-uid', interval = 5000, params = {} }) => (
  dispatch,
  getState,
) => {
  if (selectIsMonitorActive(getState(), id)) {
    return dispatch({
      type: TASKS_MONITOR_ALREADY_MONITORED,
      payload: { id, interval, params },
    });
  }

  const intervalId = setInterval(() => dispatch(runMonitorLifecycle(id)), interval);

  dispatch({
    type: TASKS_MONITOR_STARTED,
    payload: {
      id, interval, intervalId, params,
    },
  });

  // run it manually so we don't wait the interval for the first time
  return dispatch(runMonitorLifecycle(id));
};
