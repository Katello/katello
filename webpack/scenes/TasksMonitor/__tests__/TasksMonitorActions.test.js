import { testActionSnapshotWithFixtures } from '../../../move_to_pf/test-utils/testHelpers';
import { foremanTasksApi as api } from '../../../services/api';
import { selectMonitor, selectIsMonitorActive } from '../TasksMonitorSelectors';

import { monitor } from './TasksMonitor.fixtures';
import { stopMonitoringTasks, startMonitoringTasks, runMonitorLifecycle } from '../TasksMonitorActions';

jest.mock('../../../services/api');
jest.mock('../TasksMonitorSelectors');

const noop = () => null;

const fixtures = {
  'should stop monitoring tasks': {
    action: () => dispatch => stopMonitoringTasks('some-monitor-id')(dispatch, noop),
    test: () => expect(clearInterval).toHaveBeenCalledWith(monitor.intervalId),
  },
  'should start monitoring tasks': {
    action: () => (dispatch) => {
      const id = 'some-monitor-id';
      const { interval, params } = monitor;

      return startMonitoringTasks({ id, interval, params })(dispatch, noop);
    },
    test: () => expect(setInterval).toHaveBeenCalled(),
  },
  'should not start monitoring tasks if already monitored': {
    action: () => (dispatch) => {
      selectIsMonitorActive.mockImplementation(() => true);

      const id = 'some-monitor-id';
      const { interval, params } = monitor;

      return startMonitoringTasks({ id, interval, params })(dispatch, noop);
    },
    test: () => expect(setInterval).not.toHaveBeenCalled(),
  },
  'should run monitor lifecycle': {
    action: () => (dispatch) => {
      api.post.mockImplementation(async () => ({
        status: 200,
        data: [{ results: 'some-tasks' }],
      }));
      return runMonitorLifecycle('some-monitor-id')(dispatch, noop);
    },
    test: () => expect(api.post.mock.calls).toMatchSnapshot(),
  },
  'should run monitor lifecycle and fail because bad server response': () => (dispatch) => {
    api.post.mockImplementation(async () => {
      throw new Error('some error');
    });
    return runMonitorLifecycle('some-monitor-id')(dispatch, noop);
  },
  'should run monitor lifecycle and fail because unauthorized': () => (dispatch) => {
    api.post.mockImplementation(async () => ({
      status: 401,
    }));
    return runMonitorLifecycle('some-monitor-id')(dispatch, noop);
  },
};

describe('TasksMonitor actions', () => {
  beforeEach(() => {
    jest.useFakeTimers();

    selectMonitor.mockImplementation(() => monitor);
    selectIsMonitorActive.mockImplementation(() => false);
  });
  afterEach(() => {
    jest.useRealTimers();
    jest.resetAllMocks();
    jest.restoreAllMocks();
    jest.resetModules();
  });

  testActionSnapshotWithFixtures(fixtures);
});
