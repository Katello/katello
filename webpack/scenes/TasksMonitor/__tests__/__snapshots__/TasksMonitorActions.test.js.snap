// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksMonitor actions should not start monitoring tasks if already monitored 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "id": "some-monitor-id",
        "interval": 5000,
        "params": Object {
          "param1": "some-param-1",
          "param2": "some-param-2",
        },
      },
      "type": "TASKS_MONITOR_ALREADY_MONITORED",
    },
  ],
]
`;

exports[`TasksMonitor actions should run monitor lifecycle 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "id": "some-monitor-id",
        "tasks": "some-tasks",
      },
      "type": "TASKS_MONITOR_SUCCESS",
    },
  ],
]
`;

exports[`TasksMonitor actions should run monitor lifecycle 2`] = `
Array [
  Array [
    "/tasks/bulk_search",
    Object {
      "searches": Array [
        Object {
          "param_1": "some-param-1",
          "param_2": "some-param-2",
        },
      ],
    },
  ],
]
`;

exports[`TasksMonitor actions should run monitor lifecycle and fail because bad server response 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "error": [Error: some error],
        "id": "some-monitor-id",
      },
      "type": "TASKS_MONITOR_FAILED",
    },
  ],
]
`;

exports[`TasksMonitor actions should run monitor lifecycle and fail because unauthorized 1`] = `
Array [
  Array [
    [Function],
  ],
  Array [
    Object {
      "payload": Object {
        "error": [Error: Unauthorized],
        "id": "some-monitor-id",
      },
      "type": "TASKS_MONITOR_FAILED",
    },
  ],
]
`;

exports[`TasksMonitor actions should start monitoring tasks 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "id": "some-monitor-id",
        "interval": 5000,
        "intervalId": 1,
        "params": Object {
          "param1": "some-param-1",
          "param2": "some-param-2",
        },
      },
      "type": "TASKS_MONITOR_STARTED",
    },
  ],
  Array [
    [Function],
  ],
]
`;

exports[`TasksMonitor actions should stop monitoring tasks 1`] = `
Array [
  Array [
    Object {
      "payload": Object {
        "id": "some-monitor-id",
        "intervalId": 1234,
      },
      "type": "TASKS_MONITOR_STOPPED",
    },
  ],
]
`;
